src/2ls/summary_checker_nonterm.cpp

```diff
@@ -0,0 +1,566 @@
+      exprt::operandst first_linearity_check;
+      exprt::operandst second_linearity_check;
+      exprt::operandst constants_computation;
+      exprt::operandst constants;
+      exprt::operandst loopback_vars;
+      exprt::operandst loop_exit_cond;
+      exprt::operandst neg_loop_exit_cond;
+
+      property_map[property_id].location=n_it->loophead->location;
+      property_map[property_id].result=UNKNOWN;
+
+      unsigned const_number=0;
+      for(local_SSAt::objectst::const_iterator
+          o_it=SSA.ssa_objects.objects.begin();
+          o_it!=SSA.ssa_objects.objects.end();
+          o_it++)
+      {
+        ssa_domaint::phi_nodest::const_iterator p_it=
+        phi_nodes.find(o_it->get_identifier());
+        if (p_it==phi_nodes.end()) continue; // object not modified in this loop
+
+        //  first linearity check data preparation
+
+        SSA.current_unwinding-=1;
+        symbol_exprt phi_var1;
+        phi_var1=SSA.name(*o_it, local_SSAt::PHI, n_it->loophead->location);
+        ssa_local_unwinder.unwinder_rename(phi_var1,*n_it->loophead,true);
+        SSA.current_unwinding+=1;
+        symbol_exprt phi_var2;
+        phi_var2=SSA.name(*o_it, local_SSAt::PHI, n_it->loophead->location);
+        ssa_local_unwinder.unwinder_rename(phi_var2,*n_it->loophead,true);
+
+        //  works only for bitvectors
+
+        if ((phi_var2.type().id()!=ID_unsignedbv) &&
+            (phi_var2.type().id()!=ID_signedbv))
+        {
+          first_linearity_check.clear();
+          break;
+        }
+
+        symbol_exprt constk("const$k", phi_var2.type());
+        symbol_exprt const1("const$" + std::to_string(const_number++),
+                            phi_var2.type());
+
+        first_linearity_check.push_back(
+              equal_exprt(phi_var1, plus_exprt(phi_var2, const1)));
+
+        // get constants data preparation
+
+        constants_computation.push_back(
+              equal_exprt(minus_exprt(phi_var1, phi_var2), const1));
+        constants.push_back(const1);
+
+        // loopback vars data preparation
+
+        exprt init_expr;
+        exprt lb_var;
+        for (local_SSAt::nodet::equalitiest::const_iterator e_it=
+          n_it->loophead->equalities.begin();
+             e_it != n_it->loophead->equalities.end(); e_it++)
+        {
+          if (e_it->rhs().id() == ID_if &&
+              to_symbol_expr(e_it->lhs()).get_identifier()==
+              phi_var2.get_identifier())
+          {
+            const if_exprt &if_expr = to_if_expr(e_it->rhs());
+            init_expr=if_expr.false_case();
+            lb_var=if_expr.true_case();
+            //should already be renamed for inner loops
+            break;
+          }
+        }
+
+        loopback_vars.push_back(lb_var);
+        loopback_vars.push_back(init_expr);
+        loopback_vars.push_back(constk);
+      }
+
+      if (first_linearity_check.empty())  //nothing to be checked
+        continue;
+
+      neg_loop_exit_cond.push_back(lhguard);
```

Comment:
It has 340 lines. It could be separated into functions but there are some continues and breaks, pushs and pops of the solver and it is working. I know that limit is 300, but couldn't we make an exception here?